tests/spread/rockcraft/plugin-jlink/parts.yaml

```diff
@@ -0,0 +1,49 @@
+
+# Actual parts definition for the rockcraft + python cases that we are testing.
+# (this gets appended to all rockcraft.yaml files in the subdirectories in the
+# main task.yaml).
+
+version: '0.1'
+summary: A rock that bundles a Java project.
+description: A rock that bundles a Java project.
+license: GPL-3.0
+platforms:
+  amd64:
+
+parts:
+  helloworld:
+    source: helloworld
+    source-type: local
+    build-packages:
+      - openjdk-21-jdk
+      - maven
+    plugin: nil
+    build-environment:
+      - JAVA_HOME: "/usr/lib/jvm/java-21-openjdk-${CRAFT_TARGET_ARCH}"
```

Comment:
Under the suggested change about putting this into the java plugins, this would be something we could add in `get_build_environment` (which is still override-able by users if need be)

Reply:
This is a band-aid for maven. Gradle searches for Java in path. 
The solution in the case would be excluding stage directory from the search path in Java plugin.